fix: Add aria attributes to tab paragraph container [REQUIRES WINDOWS]

Zeyber opened the pull request:
PDP Part of #11637

Commit:
fix: Add aria attributes to tab paragraph container

Zeyber committed:
Merge branch 'develop' into feature/GH-11637-pdp

developpeurweb commented:
**Testing outcome:**

- No violations were found with the AMP test.
- **JAWS** - overrides custom pattern and suggests navigation with up/down arrows, they work, but then when using tab it goes to next tab and not the tab panel.
- **NVDA/VO** - reads expanded/collapsed state but then when trying to go to associated tab panel, the user needs to keep tabbing to reach it, without any indication it has been reached.

**Some thoughts:**

In this case, it would be `aria-selected` instead of `aria-expanded`, but still, the pattern doesn't make sense for SRs as it is now. Because it get's "expanded" or "selected" but then when tabbing, the next tab button it's in the order path, and there is no way for the user to know when it has landed on the tab panel corresponding to the button.

So the suggestion here is to use an **Accordion** to keep a consistent focus sequence, instead of a Tab group.

Zeyber committed:
Merge branch 'develop' into feature/GH-11637-pdp

Zeyber committed:
add css accordian only flag

developpeurweb commented:
Unfortunately, the PDP remains unchanged regarding SR compatibility, because the source order doesn't help. 

We currently have this:

![Screen Shot 2022-02-01 at 4 17 02 PM](https://user-images.githubusercontent.com/15113219/152055733-cb934c5b-ab0a-4ad6-8b82-8bf77ba3e597.png)

**It would be great** if we could have it as per screenshots below. With divs having tabindex so they can receive focus `<div tabindex="0">` so users won't skip to the following button and omit the panel's content.

![Screen Shot 2022-02-01 at 4 20 56 PM](https://user-images.githubusercontent.com/15113219/152055770-b7c1fa08-e0a0-4004-975b-bc2a1e6651fe.png)

![Screen Shot 2022-02-01 at 4 22 47 PM](https://user-images.githubusercontent.com/15113219/152056000-ea7ce1d3-2849-465b-ac34-229a075e2141.png)

Zeyber committed:
move tab panel

Zeyber committed:
Merge branch 'develop' into feature/GH-11637-pdp

Zeyber committed:
fix key interaction

Zeyber commented:
Unassigning myself so a dev who can use JAWS and NVIDIA (ie. Windows machine) can take over this PR. Currently seems to work fine on Mac with Chrome Vox and Voice Over Screen Readers but there are issues only apparent the windows readers.

Zeyber committed:
Merge branch 'develop' into feature/GH-11637-pdp

FollowTheFlo committed:
feature/GH-11637-pdp: solve merge conflicts

FollowTheFlo committed:
fix:Modified aria attr and css to tab paragraph container

fix expand/collapse and sound not working with NVDA, modified aria and introduced icon placeholder in template to have title followed by icon in DOM

FollowTheFlo committed:
Merge branch 'develop' into feature/GH-11637-pdp

FollowTheFlo committed:
fix:modified icon placeholder with span element

FollowTheFlo committed:
fix:add breaking change documentation

FollowTheFlo commented:
Note for reviewer: 
breakpointService has been removed as it was decided to apply accordion on Pdp to all screen sizes. 
Tabs and Accordion Mode can be switched using css var $useAccordionOnly.

developpeurweb reviewed (changes requested):
The accordion toggle works now cross-SR/browser. However, there are a couple of new minor issues:
1. When focusing on accordion buttons, all SRs read the panel's content as opposed to only the button label, except for the Specs button (based on PLP, maybe try using `aria-label` to remove the bloat). Likewise, since this is a temporary solution and not the final component, pls try removing `aria-owns` if this makes the bug go away, it's not needed in this case.
![2022-03-02_10-51-00](https://user-images.githubusercontent.com/15113219/156417508-bc0ff3df-b662-4caf-a9d6-cc223523ece5.jpg)

2. The icon + and - also gets vocalized. Adding `role="presentation"` to it seems to fix it.
<img width="1635" alt="Screen Shot 2022-03-02 at 12 32 53 PM" src="https://user-images.githubusercontent.com/15113219/156417748-6a277827-1987-4b2b-804e-03139b07d77d.png">

FollowTheFlo committed:
Merge branch 'develop' into feature/GH-11637-pdp
merge develop# the commit.

FollowTheFlo committed:
feature/GH-11637-pdp:fix:Keep only button title vocalized when landing and remove icon vocalization

developpeurweb committed:
Merge branch 'develop' into feature/GH-11637-pdp

developpeurweb reviewed (dismissed):
This works well for JAWS*/NVDA/VO
*Works well on JAWS+Firefox. Bugs with JAWS+Chrome (it will be addressed on a separate bug ticket if persistent).

Matejk00 committed:
Merge branch 'develop' into feature/GH-11637-pdp

pawelfras committed:
Merge branch 'develop' into feature/GH-11637-pdp

pawelfras commented on docs/migration/5_0.md:
```diff
@@ -416,12 +417,6 @@ The following keys have been added to `configurator-common.ts`:
 - key `common.navigation.shopAll` has been removed.
 - added new keys: `common.navigation.categoryNavLabel`, `common.navigation.footerNavLabel`
 
-### TabParagraphContainerComponent
```
`could:` consider to add an information that `BreakpointService` has been removed from component.

pawelfras commented on docs/migration/css-changes-in-version-5.md:
```diff
@@ -43,6 +43,7 @@ title: Changes to Styles in 5.0
 
 - wrapped everything in `%pdpTabs`inside a `> div {...}` to restore styling after template changes
 - wrapped everything in `%pdpFlat`inside a `> div {...}` to restore styling after template changes
+- `span.accordion-icon` has been added for screen reader improvments.
```
`query:` maybe it's worth to mention about new var `$useAccordionOnly`?

Matejk00 commented on projects/storefrontlib/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.html:
```diff
@@ -5,20 +5,23 @@
             component.title | cxTranslate: { param: tabTitleParams[i] | async }
           }}
+
+          <span class="accordion-icon" aria-hidden="”true”"></span>
```
Wondering if this double quote is required: `aria-hidden="”true”"`
I think we should try with `aria-hidden="true"`

FollowTheFlo replied:
thanks, looks like prettier did not catch it.

morganm58 commented on projects/storefrontstyles/scss/components/content/tab-paragraph-container/_tab-paragraph-container.scss:
```diff
@@ -1,4 +1,51 @@
+      &:before {
+        content: '+';
+        margin: 0px 15px;
+        float: right;
```
The float will cause problems with directionality. We can remove the float and position this.

FollowTheFlo replied:
it slightly modified vertical align of +/- icons, which is fixed by modifying 'bottom' property. 
Tested with resizing/zooming screen, all ok.

morganm58 commented on projects/storefrontstyles/scss/components/content/tab-paragraph-container/_tab-paragraph-container.scss:
```diff
@@ -1,4 +1,51 @@
+    width: 100%;
+    text-align: start;
+    height: 63px;
+    > span.accordion-icon {
```
add `position: absolute;` and `inset-inline-end: 0;` to span.accordion

morganm58 commented on projects/storefrontstyles/scss/components/content/tab-paragraph-container/_tab-paragraph-container.scss:
```diff
@@ -1,4 +1,51 @@
+    padding-inline-start: 4%;
+    width: 100%;
+    text-align: start;
+    height: 63px;
```
Add 'position: relative;' after line 14

morganm58 commented on projects/storefrontstyles/scss/components/content/tab-paragraph-container/_tab-paragraph-container.scss:
```diff
@@ -1,4 +1,51 @@
+@mixin accordion {
+  > button {
+    border-top: 1px var(--cx-color-light) solid;
+    font-size: 18px;
```
font-size: 18px should be change to a variable. `font-size: var(--cx-font-size, 1.125rem);`

morganm58 commented on projects/storefrontstyles/scss/components/content/tab-paragraph-container/_tab-paragraph-container.scss:
```diff
@@ -1,4 +1,51 @@
+        content: '+';
+        margin: 0px 15px;
+        float: right;
+        font-size: 33px;
```
font-size: 33px should be change to a variable. `font-size: var(--cx-font-size, 2.063rem);`

FollowTheFlo committed:
feature/GH-11637-pdp:fix:font-size and icon position to handle directionality

FollowTheFlo committed:
feature/GH-11637-pdp:docs:add useAccordionOnly var and Breakpoint service removed

FollowTheFlo committed:
Merge branch 'develop' into feature/GH-11637-pdp

FollowTheFlo commented:
@developpeurweb the committed fix from today didn't modify the a11y behavior. tested on windows nvda/jaws.
I suggest to create a new ticket specific to Chrome/JAWS as you mentioned, so we can go ahead with tickets depending of this one.

developpeurweb committed:
Merge branch 'develop' into feature/GH-11637-pdp

developpeurweb reviewed (approved):
CSS approved.
Still works for JAWS/NVDA/VO.

Zeyber commented on projects/storefrontlib/cms-components/content/tab-paragraph-container/tab-paragraph-container.component.ts:
```diff
@@ -34,8 +32,7 @@ export class TabParagraphContainerComponent implements AfterViewInit, OnInit {
   constructor(
     public componentData: CmsComponentData<CMSTabParagraphContainer>,
     protected cmsService: CmsService,
-    protected winRef: WindowRef,
```
Missing migrations for this constructor change
